src/templates/middleware.js

```diff
@@ -135,5 +137,21 @@ middleware['i18n'] = async ({ app, req, res, route, store, redirect, isHMR }) =>
     }
   }
 
-  await switchLocale(locale = routeLocale ? routeLocale : locale)
+  // if (!routeLocale && route.path)
+  console.log('route.path', route.path)
+  console.log('app.i18n.locales', app.i18n.locales)
+
+  const routePathLang = route.path.slice(1).split('/').shift()
+  console.log('routePathLang', routePathLang)
+
+  if (routePathLang && routePathLang.length === 2) {
+    console.log('REDIRECTING TO ' + (route.path.slice(3) || '/'))
+    locale = routePathLang
+    app.i18n.switchLocale(locale)
```

Comment:
"app.i18n.switchLocale is not a function"

Reply:
🤔 hmm that could be explain why it does not switch on the root route

Reply:
![2018-12-07 16_30_22-start](https://user-images.githubusercontent.com/17849432/49656346-6b99f100-fa3d-11e8-860a-ec21ec8d7239.png)
I recommend using an IDE that is highlighting such errors in advance :)

Reply:
@Magnum5234 which editor are you using?

Reply:
I use fall all web related stuff PHP Storm
https://www.jetbrains.com/phpstorm/